src/WebHelpers.bas

```diff
@@ -513,25 +544,90 @@ End Sub
 ' @param {WebResponse} Response
 ''
 Public Sub LogResponse(Client As WebClient, Request As WebRequest, Response As WebResponse)
-    If EnableLogging Then
-        Dim web_KeyValue As Dictionary
-
-        Debug.Print "<-- Response - " & Format(Now, "Long Time")
-        Debug.Print Response.StatusCode & " " & Response.StatusDescription
+    If EnableLogging Or EnableFileLogging Then
+        Dim msg As String
+        msg = "<-- Response - " & Format(Now, "Long Time") & vbNewLine
+        msg = msg & Response.StatusCode & " " & Response.StatusDescription & vbNewLine
 
+        Dim web_KeyValue As Dictionary
         For Each web_KeyValue In Response.Headers
-            Debug.Print web_KeyValue("Key") & ": " & web_KeyValue("Value")
+            msg = msg & web_KeyValue("Key") & ": " & web_KeyValue("Value") & vbNewLine
         Next web_KeyValue
 
         For Each web_KeyValue In Response.Cookies
-            Debug.Print "Cookie: " & web_KeyValue("Key") & "=" & web_KeyValue("Value")
+            msg = msg & "Cookie: " & web_KeyValue("Key") & "=" & web_KeyValue("Value") * vbNewLine
         Next web_KeyValue
 
-        Debug.Print vbNewLine & Response.Content & vbNewLine
+        msg = msg & vbNewLine & Response.Content & vbNewLine
+    End If
+    If EnableLogging Then
+        Debug.Print msg
+    End If
+    If EnableLogging Or EnableFileLogging Then
+        LogWrite "D", msg, "WebHelpers.LogResponse"
+    End If
+End Sub
+
+''
+' Close the log file if open.
+'
+''
+Public Sub LogClose(Optional ByVal Dummy as Boolean)
+    If LogFile <> "" And LogFileNumber > 0 Then
+        Close #LogFileNumber
+        LogFileNumber = 0
     End If
 End Sub
 
 ''
+' Log a message to a log file if specified.
+'
+' @param {String} LogType (E=Error, W=Warning, D=Debug)
+' @param {String} Message
+' @param {String} From
+''
+Private Sub LogWrite(ByVal LogType As String, ByVal Message As String, ByVal From As String)
+    If LogFile <> "" Then
+        If LogFileNumber < 0 Then
+            LogFileNumber = LogFileNumber + 1
+            Exit Sub
+        End If
+
+        If LogFileNumber = 0 Then
+            On Error GoTo FileOpenError
+            LogFileNumber = FreeFile
+            Open LogFile For Append Access Write Lock Write As #LogFileNumber
+            On Error GoTo 0
+        End If
+
+        Dim Lines() As String
+        Dim msg As Variant
+        Lines = split(Message, vbNewLine)
+        On Error GoTo FileWriteError
+        For Each msg In Lines
+            msg = Replace(msg, """", "'")
+            If Trim(msg) <> "" Then _
+                Write #LogFileNumber, LogType, Format(Now, "General Date"), From, Trim(msg)
+        Next msg
+    End If
+    Exit Sub
+
+FileOpenError:
+    Debug.Print "ERROR: Unable to open logfile '" & LogFile & "' for Append Write: Error " & Err.Number & ": " & Err.Description
+    LogFileNumber = -100
```

Comment:
The idea here is not to issue excessive numbers of these errors to the immediate window - but upon reflection I am not sure this is such a good idea - or perhaps only if normal EnableLogging is true.